update 115 (#172)
tari-project/tari-dot-com: _updates/2023-08-10-update-115.md

---
layout: update
tag: Developer Update
date: 2023-08-10
author: swvheerden
thumbnail: update-background.jpg
title: Network rollover
class: subpage
---

> The [Tari network upgrade schedule](..link please) calls for a network rollover every 8 weeks. This week marks a designated rollover juncture, which means that Nextnet will transition to Stagenet, and Development will move to Nextnet. These rollovers do not typically entail a hardfork. This time, however, all three test networks will be reset with a fresh Genesis block as we roll out some new features.

Reviewer: ```suggestion
The [Tari network upgrade schedule](https://github.com/tari-project/tari/blob/development/Contributing.md#the-minotari-release-process) calls for a network rollover every 8 weeks. This week marks a designated rollover juncture, which means that Nextnet will transition to Stagenet, and Development will move to Nextnet. These rollovers do not typically entail a hardfork. This time, however, all three test networks will be reset with a fresh Genesis block as we roll out some new features.
```

---
layout: update
tag: Developer Update
date: 2023-08-10
author: swvheerden
thumbnail: update-background.jpg
title: Network rollover
class: subpage
---

> As you may be aware, we have a regular network rollover every 8 weeks. This week is the designated rollover period, which means that Nextnet will transition to Stagenet, and Development will move to Nextnet. With these transitions, it's important to note that each network will undergo a reset. Therefore, we kindly remind you to clean up your network folders accordingly.

Reviewer: ```suggestion
The [Tari network upgrade schedule](..link please) calls for a network rollover every 8 weeks. This week marks a designated rollover juncture, which means that Nextnet will transition to Stagenet, and Development will move to Nextnet. These rollovers do not typically entail a hardfork. This time, however, all three test networks will be reset with a fresh Genesis block as we roll out some new features.
```

> The primary focus of these changes is centered around bolstering security measures for Tari, ensuring a robust and bug-free environment before our official launch. Our team has been diligent in conducting an internal audit, addressing potential vulnerabilities, and closing any identified loopholes. Concurrently, we are engaging in an external audit to further enhance the security of our network.

Reviewer: ```suggestion
```

## Stagenet

> Let's highlight some of the noteworthy changes on Stagenet:

Reviewer: ```suggestion
The notable highlights of the upcoming changes to Stagenet are:
```

*   make the first output optional in the wallet (#5352) ([bf16140e](https://github.com/tari-project/tari/commit/bf16140ecd1ad0ae25f8a9b8cde9c3e4f1d12a02))
*   wallet .h file for lib wallets (#5330) ([22a3a17d](https://github.com/tari-project/tari/commit/22a3a17db6ef8889cb3a73dfe2db081a0691a68c))
*   clippy issues with config (#5334) ([026f0d5e](https://github.com/tari-project/tari/commit/026f0d5e33d524ad302e7edd0c82e108a17800b6)
*   default network selection (#5333) ([cf4b2c8a](https://github.com/tari-project/tari/commit/cf4b2c8a4f5849ba51dab61595dfed1a9249c580)
*   allow public addresses from command line (#5303) ([349ac895](https://github.com/tari-project/tari/commit/349ac8957bc513cd4110eaac69550ffa0816862b)
* **comms:**
  *  validate onion3 checksum (#5440) ([0dfdb3a4](https://github.com/tari-project/tari/commit/0dfdb3a4bef51952f0cecf6f6fcb00f6b2bfe302)
  *  only set final forward address if configured to port 0 (#5406) ([ff7fb6d6](https://github.com/tari-project/tari/commit/ff7fb6d6b4ab4f77d108b2d9b7fd010c77e613c7)
* **wallet-ffi:**  don't block on start (#5437) ([27fe8d9d](https://github.com/tari-project/tari/commit/27fe8d9d2fc3ea6468605ef5edea56efdcc8248f)

## Nextnet

> During this period, we have prioritized a rigorous internal audit and devoted significant resources to address security concerns. We've diligently tackled a wide range of bugs, addressing issues from refining the mempool's handling of zero-confirmation transactions to resolving malleability bugs present in the headers.

Reviewer: ```suggestion
During this period, a wide range of bugs, addressing issues from refining the mempool's handling of zero-confirmation transactions to resolving malleability bugs present in the headers, have been addressed.
```

> A notable improvement in our wallet recovery process is worth highlighting. We've transitioned away from the practice of rewinding the bulletproof, opting for a more streamlined approach: now, the blinding factor and value on the utxo are simply encrypted, enhancing the efficiency of the recovery mechanism.

Reviewer: ```suggestion
A notable improvement in the wallet recovery process is worth highlighting. There is a transition away from the practice of rewinding the bulletproof, opting for a more streamlined approach: now, the blinding factor and value on the UTXO are simply encrypted, enhancing the efficiency of the recovery mechanism.
```

> Additionally, we've introduced a noteworthy enhancement to the Tari layer - the incorporation of a chat FFI client tailored specifically for peer-to-peer (p2p) chat messaging. This addition significantly enhances the communication capabilities within the Tari network, enriching the overall user experience.

Reviewer: ```suggestion
Additionally, there is a noteworthy enhancement to the Tari layer - the incorporation of a chat FFI client tailored specifically for peer-to-peer (p2p) chat messaging. This addition significantly enhances the communication capabilities within the Tari network, enriching the overall user experience.
```

The full list is available below

#### Features

> *****Add finalized list when done

Reviewer: ^^ TODO
